### PR TITLE
Make all non-required attributes nullable

### LINKED ISSUE #3958: Make all non-required attributes nullable.
Allow, `null` value for all non-required attributes across all endpoints. 
Make necessary model, db changes. (Don't forget to include proper migrations) 

(Exceptions: boolean)

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #3958 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 

#### Short description of what this resolves:
Allows all non-required attributes accept `null` in API requests

#### Changes proposed in this pull request:

- Adds `allow_none` as per marshmallow fields specification.

### CODE REVIEW

**Comment by enigmaeth** [comment]
@niranjan94 They are accepting null values by default( since `nullable=False` is not set). 
The API requests gave error because flask-rest-jsonapi raises validation error on `null` values. To avoid which I have added `allow_none=True". [[marshmallow fields](http://marshmallow.readthedocs.io/en/latest/api_reference.html#module-marshmallow.fields)]

**Comment by enigmaeth** [comment]
@niranjan94 @SaptakS Please review. Thanks!

**Comment by enigmaeth** [comment]
@niranjan94 Please take a look at this also :sweat_smile:

File: app/api/event_invoices.py [hunk 1]
```diff
@@ -26,27 +26,28 @@ class Meta:
...
-    identifier = fields.Str()
-    amount = fields.Float(validate=lambda n: n >= 0)
-    address = fields.Str()
-    city = fields.Str()
-    state = fields.Str()
-    country = fields.Str(validate=validate.OneOf(choices=PAYMENT_COUNTRIES))
-    zipcode = fields.Str()
-    created_at = fields.DateTime()
+    identifier = fields.Str(allow_none=True)
+    amount = fields.Float(validate=lambda n: n >= 0, allow_none=True)
```
Comment (arp95) [inline app/api/event_invoices.py]: whats the point of having an identifier when we have an id for the event?

File: app/api/sessions.py [hunk 2]
```diff
@@ -49,22 +49,23 @@ def validate_date(self, data, original_data):
...
+    signup_url = fields.Url(allow_none=True)
+    state = fields.Str(validate=validate.OneOf(choices=["pending", "accepted", "confirmed", "rejected", "draft"]),
+                       allow_none=True)
     created_at = fields.DateTime(dump_only=True)
     deleted_at = fields.DateTime(dump_only=True)
-    submitted_at = fields.DateTime()
+    submitted_at = fields.DateTime(allow_none=True)
     is_mail_sent = fields.Boolean()
     microlocation = Relationship(attribute='microlocation',
                                  self_view='v1.session_microlocation',
```
Comment (arp95) [inline app/api/sessions.py]: cant there be some cases when the location for a session be null?
like an online session or something.

**Comment by shubham-padia** [comment]
@arp95 Please do not ask general doubts like `why do event identifiers exist` on the PRs :smile:. Feel free to ask any query on the server gitter channel itself.

**Pull request merged**